Polygon_mesh_processing/examples/Polygon_mesh_processing/non_rigid_mesh_registration_example.cpp

```diff
@@ -0,0 +1,88 @@
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h>
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/Polygon_mesh_processing/IO/polygon_mesh_io.h>
+
+#include <CGAL/Point_set_3.h>
+
+#include <CGAL/Timer.h>
+
+using K = CGAL::Exact_predicates_inexact_constructions_kernel;
+using Mesh = CGAL::Surface_mesh<K::Point_3>;
+using Point = K::Point_3;
+using Vector = K::Vector_3;
+using FT = K::FT;
+
+namespace PMP = CGAL::Polygon_mesh_processing;
+
+int main(int argc, char** argv) {
+  const std::string source_fn = CGAL::data_file_path("meshes/bear.off");
+  const std::string target_fn = CGAL::data_file_path("meshes/bear_bis.off");
+  const std::string corr_fn = "data/bear_bear_bis.corr";
+
+  Mesh source, target;
+  if (!PMP::IO::read_polygon_mesh(source_fn, source))
+  {
+    std::cerr << "Invalid input " << source_fn << std::endl;
+    return 1;
+  }
+
+  if (!PMP::IO::read_polygon_mesh(target_fn, target))
+  {
+    std::cerr << "Invalid input " << target_fn << std::endl;
+    return 1;
+  }
+
+  std::vector<std::pair<Mesh::Vertex_index, Mesh::Vertex_index>> correspondences_mesh;
+  std::ifstream corr(corr_fn);
+  if (corr.is_open()) {
+    std::string line;
+    while (std::getline(corr, line)) {
+      if (line.size() == 0) continue;
+      std::istringstream iss(line);
+
+      std::size_t v0, v1;
+      if (iss >> v0 >> v1)
+        Mesh source, target;
+      correspondences_mesh.push_back(std::make_pair(*(source.vertices_begin() + v0), *(target.vertices_begin() + v1)));
+    }
+
+    std::cout << correspondences_mesh.size() << " correspondences provided" << std::endl;
+  }
+
+  auto vnm = target.add_property_map<Mesh::Vertex_index, K::Vector_3>("v:normal");
+  if (vnm.second)
+    PMP::compute_vertex_normals(target, vnm.first);
+
+  typename Mesh::Property_map<Mesh::Vertex_index, CGAL::Aff_transformation_3<K>> vrm = source.add_property_map<Mesh::Vertex_index, CGAL::Aff_transformation_3<K>>("v:rotation").first;
+  typename Mesh::Property_map<Mesh::Vertex_index, K::Vector_3> vtm = source.add_property_map<Mesh::Vertex_index, K::Vector_3>("v:transformations").first;
```

Comment:
```suggestion
  typename Mesh::Property_map<Mesh::Vertex_index, K::Vector_3> vtm = source.add_property_map<Mesh::Vertex_index, K::Vector_3>("v:translation").first;
```

Reply:
I do not understand why we need two affine transformations.   An single one can combine rotation and translation. In general it is a 3x4 matrix where the first three columns are the rotation and the last one the translation.  
Another question is if it should be a quaternion and a Vector_3  instead. Yes, we do not have that in CGAL yet.